test/unit/stop_test.lua

```diff
@@ -0,0 +1,34 @@
+local expirationd = require("expirationd")
+local fiber = require("fiber")
+local t = require("luatest")
+local g = t.group("stop")
+
+local helpers = require("test.helper")
+
+g.before_all(function()
+    helpers.init_spaces(g)
+end)
+
+g.after_each(function()
+    helpers.truncate_spaces(g)
+end)
+
+function g.test_cancel_on_pcall()
+    local function on_full_scan_complete()
+        pcall(function() fiber.sleep(1) end)
```

Comment:
nit: Could be rewritten as pcall(fiber.sleep, 1)

Reply:
Yes, that would be more correct. Fixed